edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudProxy.cs

```diff
@@ -152,12 +153,18 @@ public async Task SendMessageAsync(IMessage inputMessage)
                 string outputRoute = inputMessage.GetOutput();
                 using (Metrics.TimeMessageSend(this.clientId, outputRoute))
                 {
+                    var tokenSource = new CancellationTokenSource(this.sdkWaitTime);
                     Metrics.MessageProcessingLatency(this.clientId, inputMessage);
-                    await this.client.SendEventAsync(message);
+                    await this.client.SendEventAsync(message, tokenSource.Token).TimeoutAfterSDKHang(TimeSpan.FromSeconds(50));
```

Comment:
Do we only need to worry about this for SendEvent? What about Twin operations, etc?

Reply:
So while we haven't seen this issue happen with other operations, since its still being investigated in the SDK, we arent exactly sure if it could occur in situations other than SendEvent.

Reply:
With the total possibility that this sdk issue could occur and us not being sure, went ahead and added the task extension to additional operations. I also moved the logic to handle the error to the HandleException task to reduce duplicate logic